### ISSUE TITLE
Document process of merging `nmdc-schema/main` into `berkeley-schema-fy24/main` (in local slang: "backmerge")

### ISSUE THREAD

**Issue by eecavanna** [microbiomedata/nmdc-schema#2170]
Document process of merging `nmdc-schema/main` into `berkeley-schema-fy24/main` (in local slang: "backmerge", "back-merge", "back merge").

Although I don't expect myself to be the only author here, there are some things I already want to document about the process. I am creating this ticket to act as a temporary home for that documentation.

**Comment by eecavanna** [comment]
#### 1. Create a branch in the fork

1. Go to https://github.com/microbiomedata/berkeley-schema-fy24/branches
2. Click the "New branch" button (upper right)
3. Fill in the form
  1. Name the branch something like: `berkeley-main-clone`
  2. Set the source repo to: `https://github.com/microbiomedata/berkeley-schema-fy24`
  3. Set the source branch to: `main`
4. Click "Create new branch"

#### 2. Create a PR in the fork

1. Go to https://github.com/microbiomedata/nmdc-schema/pulls
2. Click the "New pull request" button (upper right)
3. Click the words "compare across forks" near the top of the page
4. Set the branches as follows:
   1. Base repository: `microbiomedata/berkeley-schema-fy24`
   2. Base (branch): `berkeley-main-clone` (or whatever you called the newly-created branch)
   3. Head repository: `microbiomedata/nmdc-schema`
   4. Compare (branch): `main`
5. Click the "Create pull request" button
6. Fill in the title field:
   ```
   Merge `nmdc-schema/main` into copy of `berkeley-schema-fy24/main` (backmerge YYYY-MM-DD)
   ```
   > Replace `YYYY-MM-DD` with today's date
7. Next to the "Create pull request" button, click the 🔽 "down arrow" icon and select "Create draft pull request"
8. Click the "Draft pull request" button

#### 3. Resolve merge conflicts

TODO

**Comment by eecavanna** [comment]
I will work with @turbomam on filling in section 3 of the above guide.

**Comment by eecavanna** [comment]
I think @mslarae13 also wants the following to be documented:
- Under what conditions someone will create a pre-release
- How to create a pre-release

**Comment by eecavanna** [comment]
We tested out [these steps](https://github.com/microbiomedata/nmdc-schema/issues/2170#issuecomment-2307798931) today. At step 3, we did not end up with a branch where we could collaborate; so, this didn't put is in any "better" position than before.

**Comment by eecavanna** [comment]
> I think @mslarae13 also wants the following to be documented:
>
> - Under what conditions someone will create a pre-release
> - How to create a pre-release

Here is my proposal for this:

Any time the `main` branch of the `berkeley-schema-fy24` repo has changed since the last time a pre-release was created, anyone that has sufficient access on GitHub can create a new pre-release of it, provided they follow all the relevant steps in the https://github.com/microbiomedata/infra-admin/blob/main/releases/nmdc-schema.md guide, plus they mark the "Pre-release" checkbox after doing the existing steps in the "Making a Release > Create a new GitHub Release" section of that guide.

In short, in the context of the Berkeley schema and the stage of the Roll Out we're in, my view is that "no change is too small to warrant a release" and "no time is too soon to do another release."

**Comment by eecavanna** [comment]
> Document process of merging `nmdc-schema/main` into `berkeley-schema-fy24/main` (in local slang: "backmerge", "back-merge", "back merge").

My proposal for this original task is:

Given that @turbomam is the only person that happens to have done this process end-to-end so far, we wait until the next time he is going to go through the process, and then either he document it as he goes through it, or he pair up with me or someone else to document it (or at least note the steps) at that time.

**Comment by turbomam** [comment]
- @turbomam to add the following (in google doc)
- @mslarae13 to suggest a google drive path
    - it doesn't appear possible to make a branch (for multiple people to access) 
    - so the backmerge is best done with two or three people who have a combination of domain, modelling and GitHub skills
    - delete generated files before resolving conflicts
    - there are more schema files in berkeley-schema-fy24  than there are in nmdc-schema. be prepared to look for content that is duplicate in two files after the merge
    - schema conflicts should be worked though with the guidance of example data files and vice versa

**Comment by ssarrafan** [comment]
@turbomam @mslarae13 @eecavanna is this documentation task complete?

**Comment by eecavanna** [comment]
Not yet. I don't expect this to be finished this sprint/week.

Our most recent plan was to wait until the next time @turbomam performs the steps, and document them at that time. 

Also, given that the fork-to-upstream merge is scheduled to happen on Monday, I don't know when @turbomam would be performing these steps next, if ever.

Once the Berkeley Schema Roll Out has concluded, I'm planning to work with @turbomam to make it so derivative files don't get stored in the repo; which I think would change (simplify) the process this ticket is about documenting.

**Comment by ssarrafan** [comment]
I'll move to sprint 48.

**Comment by ssarrafan** [comment]
Removing from sprint. No updates in a month. @turbomam @eecavanna @mslarae13

**Comment by mslarae13** [comment]
@eecavanna @turbomam 

Is there still effort to do this?
I know there's been some documentation, and if we don't think we need this anymore, we can close as will not complete. 

(Trying to wrap up this squad's project board)

**Comment by eecavanna** [comment]
I am in favor of closing it.

**Comment by eecavanna** [comment]
> Once the Berkeley Schema Roll Out has concluded, I'm planning to work with @turbomam to make it so derivative files don't get stored in the repo; which I think would change (simplify) the process this ticket is about documenting.

I'm still planning to do what I described in that quote—just haven't gotten to it yet 🙁 .

**Comment by mslarae13** [comment]
@eecavanna @turbomam I'd like to close out the berkeley schema project board. This is still around. Could we expect this to be wrapped up sometime soon?

**Comment by eecavanna** [comment]
I discussed this ticket briefly with @mslarae13 via Slack and we decided to move it out of the project board, given that has been idle for a couple months and its incompletion was blocking her from (conceptually) closing out the project board. So, it is now on the "Non-squad board."
